react-hot-button/my-app/src/App.tsx

```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
+import './App.css';
+import { HotButton } from './HotButton';
+
+function App() {
+  const [count, setCount] = useState(0);
```

Comment:
This implementation is fine. You could also put the state inside the HotButton. There are advantages and disadvantages to each approach. Generally, I prefer to make my components as self-contained as possible.